src/Symfony/Component/JsonEncoder/DependencyInjection/JsonEncoderPass.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\JsonEncoder\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\JsonEncoder\DecoderInterface;
+use Symfony\Component\JsonEncoder\EncoderInterface;
+
+/**
+ * Injects encodable classes into services and registers aliases.
+ *
+ * @author Mathias Arlaud<mathias.arlaud@gmail.com>
+ */
+final readonly class JsonEncoderPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        if (!$container->hasDefinition('json_encoder.encoder')) {
+            return;
+        }
+
+        $container->registerAliasForArgument('json_encoder.encoder', EncoderInterface::class, 'json.encoder');
+        $container->registerAliasForArgument('json_encoder.decoder', DecoderInterface::class, 'json.decoder');
+
+        $encodablePaths = $container->hasParameter('json_encoder.encodable_paths') ? $container->getParameter('json_encoder.encodable_paths') : [];
+        $encodableClassNames = $this->getEncodableClassNames($encodablePaths);
+
+        $container->getDefinition('.json_encoder.cache_warmer.encoder_decoder')
+            ->replaceArgument(0, $encodableClassNames);
```

Comment:
`getEncodableClassNames` returns a Generator. This won't work here as a Generator cannot be dumped in the compiled container.